universe/syncer.go

```diff
@@ -165,9 +169,25 @@ func (s *SimpleSyncer) syncRoot(ctx context.Context, remoteRoot BaseRoot,
 	log.Tracef("UniverseRoot(%v): diff_size=%v, diff=%v", uniID.String(),
 		len(keysToFetch), spew.Sdump(keysToFetch))
 
+	// Before we start fetching leaves, we already start our batch stream
+	// for the new leaves. This allows us to stream the new leaves to the
+	// local registrar as they're fetched.
+	var (
+		fetchedLeaves = make(chan *IssuanceItem, len(keysToFetch))
+		newLeafProofs []*MintingLeaf
+		batchSyncEG   errgroup.Group
+	)
+
+	// We use an error group to simply the error handling of a goroutine.
+	batchSyncEG.Go(func() error {
+		newLeafProofs, err = s.batchStreamNewItems(
```

Comment:
Very cool pattern! I think we might need just one more light layer of abstraction to make it a bit more readable, but I dig the structured concurrency here.